ecmd-core/capi/ecmdTraceDefines.H

```diff
@@ -28,63 +34,8 @@
-#define ETRAC5(fmt, arg1, arg2, arg3, arg4, arg5) printf( "%s> ETRC: " fmt "\n", __FUNCTION__, arg1, arg2, arg3, arg4, arg5);
-#define ETRAC6(fmt, arg1, arg2, arg3, arg4, arg5, arg6) printf( "%s> ETRC: " fmt "\n", __FUNCTION__, arg1, arg2, arg3, arg4, arg5, arg6);
-#define ETRAC7(fmt, arg1, arg2, arg3, arg4, arg5, arg6, arg7) printf( "%s> ETRC: " fmt "\n", __FUNCTION__, arg1, arg2, arg3, arg4, arg5, arg6, arg7);
-#define ETRAC8(fmt, arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8) printf( "%s> ETRC: " fmt "\n", __FUNCTION__, arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8);
-#define ETRAC9(fmt, arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8, arg9) printf( "%s> ETRC: " fmt "\n", __FUNCTION__, arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8, arg9);
-
-#define PTRAC0(fmt) \
-  {char buffer [255]; \
-  snprintf( buffer, 255, "%s> PTRC: " fmt, __FUNCTION__); \
-  ecmdOutput(buffer);}
-#define PTRAC1(fmt, arg1) \
-  {char buffer [255]; \
-  snprintf( buffer, 255, "%s> PTRC: " fmt, __FUNCTION__, arg1); \
-  ecmdOutput(buffer);}
-#define PTRAC2(fmt, arg1, arg2) \
-  {char buffer [255]; \
-  snprintf( buffer, 255, "%s> PTRC: " fmt, __FUNCTION__, arg1, arg2); \
-  ecmdOutput(buffer);}
-#define PTRAC3(fmt, arg1, arg2, arg3) \
-  {char buffer [255]; \
-  snprintf( buffer, 255, "%s> PTRC: " fmt, __FUNCTION__, arg1, arg2, arg3); \
-  ecmdOutput(buffer);}
-#define PTRAC4(fmt, arg1, arg2, arg3, arg4) \
-  {char buffer [255]; \
-  snprintf( buffer, 255, "%s> PTRC: " fmt, __FUNCTION__, arg1, arg2, arg3, \
-  arg4); \
-  ecmdOutput(buffer);}
-#define PTRAC5(fmt, arg1, arg2, arg3, arg4, arg5) \
-  {char buffer [255]; \
-  snprintf( buffer, 255, "%s> PTRC: " fmt, __FUNCTION__, arg1, arg2, arg3, arg4, \
-  arg5); \
-  ecmdOutput(buffer);}
-#define PTRAC6(fmt, arg1, arg2, arg3, arg4, arg5, arg6) \
-  {char buffer [255]; \
-  snprintf( buffer, 255, "%s> PTRC: " fmt, __FUNCTION__, arg1, arg2, arg3, arg4, \
-  arg5, arg6); \
-  ecmdOutput(buffer);}
-#define PTRAC7(fmt, arg1, arg2, arg3, arg4, arg5, arg6, arg7) \
-  {char buffer [255]; \
-  snprintf( buffer, 255, "%s> PTRC: " fmt, __FUNCTION__, arg1, arg2, arg3, arg4, \
-  arg5, arg6, arg7); \
-  ecmdOutput(buffer);}
-#define PTRAC8(fmt, arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8) \
-  {char buffer [255]; \
-  snprintf( buffer, 255, "%s> PTRC: " fmt, __FUNCTION__, arg1, arg2, arg3, arg4, \
-  arg5, arg6, arg7, arg8); \
-  ecmdOutput(buffer);}
-#define PTRAC9(fmt, arg1, arg2, arg3, arg4, arg5, arg6, arg7, arg8, arg9) \
-  {char buffer [255]; \
-  snprintf( buffer, 255, "%s> PTRC: " fmt, __FUNCTION__, arg1, arg2, arg3, arg4, \
-  arg5, arg6, arg7, arg8, arg9); \
-  ecmdOutput(buffer);}
-
+// This define sticks an ERROR and function name at the start, then uses the
+// errmsg and varargs passed in to print the message
+#define ETRAC(errmsg, ...) printf("ERROR: (%s): " errmsg, __PRETTY_FUNCTION__, ##__VA_ARGS__);
```

Comment:
@thejta does __PRETTY_FUNCTION__ work with xlc?
I think this was used in other repos
```
#ifdef __GNUC__
#define FUNCNAME methodName(__PRETTY_FUNCTION__)
#else
#define FUNCNAME methodName(__func__)
#endif
```

Reply:
Good point on aix - I had that thought at one point and lost track of it.  Would you rather it just be `__FUNCTION__` so that it's consistent between the environments?

Reply:
yeah, I guess `__FUNCTION__` would be more portable.

Reply:
Okay - will change it.

Reply:
This is now the error messsage:
`ERROR: (flipBit): cannot flip non-binary data at bit 1 `